src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,814 @@
+    // - - - - - - - - - - - - - DOM read operations - - - - - - - - - - - - -
+    const computedStyle = getComputedStyle(this.el);
+
+    // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_START_SIZE,
+      computedStyle.borderBlockStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_BLOCK_END_SIZE,
+      computedStyle.borderBlockEndWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_START_SIZE,
+      computedStyle.borderInlineStartWidth
+    );
+
+    this.el.style.setProperty(
+      DIALOG_BORDER_INLINE_END_SIZE,
+      computedStyle.borderInlineEndWidth
+    );
+  };
+
+  #removeBorderSizeWatcher = () => {
+    if (this.#borderSizeObserver) {
+      this.#borderSizeObserver.disconnect();
+      this.#borderSizeObserver = null; // Free the memory
+    }
+
+    this.#borderSizeRAF = null; // Free the memory
+  };
+
+  #fixDialogPosition = () => {
+    const dialogRect = this.#dialogRef.getBoundingClientRect();
+    const inlineStart = this.#isRTLDirection
+      ? document.documentElement.getBoundingClientRect().width -
+        (dialogRect.left + dialogRect.width)
+      : dialogRect.left;
+    const blockStart = dialogRect.top;
+
+    setProperty(
+      this.#dialogRef,
+      DIALOG_INLINE_START,
+      inlineStart - this.#draggedDistanceX
+    );
+    setProperty(
+      this.#dialogRef,
+      DIALOG_BLOCK_START,
+      blockStart - this.#draggedDistanceY
+    );
+  };
+
+  #unfixDialogPosition = () => {
+    this.#dialogRef.style.removeProperty(DIALOG_INLINE_START);
+    this.#dialogRef.style.removeProperty(DIALOG_BLOCK_START);
+  };
+
+  render() {
+    const dialogParts = this.showHeader ? "dialog has-header" : "dialog";
+
+    return (
+      <Host
+        class={{
+          "gx-dialog-header-drag": !this.hidden && this.allowDrag === "header",
+          [RESIZING_CLASS]: this.resizing
+        }}
+      >
+        <dialog
+          aria-labelledby={this.caption ? "heading" : null}
+          part={dialogParts}
+          onClose={this.#handleDialogToggle}
+          ref={el => (this.#dialogRef = el)}
+          onMouseDown={this.allowDrag === "box" ? this.#handleMouseDown : null}
+        >
+          {this.showHeader && (
+            <header
+              class="header"
+              part="header"
+              onMouseDown={
+                this.allowDrag === "header" ? this.#handleMouseDown : null
+              }
+            >
+              <slot name="header-start" />
+              {this.caption && (
+                <h2 id="heading" part="caption" class="caption">
+                  {this.caption}
+                </h2>
+              )}
+              <slot name="header-end" />
+              <button
+                part="close"
+                class="close-button"
+                onClick={this.#closeHandler}
+              ></button>
+            </header>
```

Comment:
We should rethink the solution to add more configuration for the header. At the moment, two slots doesn't seem like the best option, so we should remove it and see how we can model it in the future.

Also, header elements doesn't work inside dialog elements, since the dialog element does not define sectioning content. See for reference:
https://html.spec.whatwg.org/#sectioning-content

```suggestion
            <div
              class="header"
              part="header"
              onMouseDown={
                this.allowDrag === "header" ? this.#handleMouseDown : null
              }
            >
              {this.caption && (
                <h2 id="heading" part="caption" class="caption">
                  {this.caption}
                </h2>
              )}

              <button
                class="close-button"
                part="close-button"
                type="button"
                onClick={this.#closeHandler}
              ></button>
            </div>
```

Reply:
Ok, thanks. I think a slot could come in handy for some cases. For example, for adding a dropdown menu like the following:

![image](https://github.com/genexuslabs/chameleon-controls-library/assets/54677414/5ba8193e-7139-4a13-b354-f1936c8f2311)